### PR TITLE
WholeGraph Feature Store for cuGraph-PyG and cuGraph-DGL

### PR DESCRIPTION
**Pull request by alexbarghi-nv**
Created based on code from @dongxuy04 

Adds support for `WholeGraph` `WholeMemory` in the cuGraph `FeatureStore` class.  This enables both DGL and PyG to take advantage of distributed feature store functionality.

Adds `pylibwholegraph` as a testing dependency so the feature store can be tested.  Adds appropriate SG and MG tests.

### CODE REVIEW

**Review by raydouglass — CHANGES_REQUESTED** [review]
`update-version.sh` needs to be updated for this new dependency. Probably just adding it to the list [here](https://github.com/rapidsai/cugraph/blob/branch-23.10/ci/release/update-version.sh#L81).

**Review by raydouglass — APPROVED** [review]

**Review by BradReesWork — APPROVED** [review]

File: python/cugraph/cugraph/gnn/feature_storage/feat_storage.py [hunk 1]
```diff
@@ -104,6 +133,31 @@ def _cast_feat_obj_to_backend(feat_obj, backend: str):
                 return _cast_to_torch_tensor(feat_obj.values)
             else:
                 return _cast_to_torch_tensor(feat_obj)
+        elif backend == "wholegraph":
+            wg_comm_obj = kwargs.get("wg_comm", wgth.get_local_node_communicator())
+            wg_type_str = kwargs.get("wg_type", "distributed")
+            wg_location_str = kwargs.get("wg_location", "cuda")
```
Comment (VibhuJawa) [inline python/cugraph/cugraph/gnn/feature_storage/feat_storage.py:139]: I think we should  store these while initiating the FeatureStore object as I don't expect these to change b/w function calls.

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Requested a minot change around API

File: python/cugraph/cugraph/gnn/feature_storage/feat_storage.py (hunk 1 above)
Comment (alexbarghi-nv) [reply python/cugraph/cugraph/gnn/feature_storage/feat_storage.py]: @VibhuJawa is it better now?

**Review by VibhuJawa — APPROVED** [review]
LGTM

**Comment by alexbarghi-nv** [comment]
This PR is going to be stuck until we have 23.10 pip packages for WholeGraph

**Pull request merged**